### ISSUE TITLE
Is this still alive?

### ISSUE THREAD

**Issue by d8ahazard** [marcosscriven/galeforce#52]
Hey there. I've got a few google wifi points, plus one of the new nest APs. I'm interested in using GF for my devices to try and add more functionality, but looking at the discussions and comments, I don't see anything recent that confirms this still works.

Could you please confirm that this is still useable in 2021, or has some issue caused it to not work anymore?

Either way, your work is greatly appreciated. Hope to hear back.

**Comment by darthnithin** [comment]
It would appear that maybe it works judging from https://github.com/marcosscriven/galeforce/pull/47#issuecomment-704491729

**Comment by ViPeR5000** [comment]
News?

**Comment by Dean2706** [comment]
Any confirmation yet? I can see that the discontinued message has been taken down.

**Comment by d8ahazard** [comment]
Maybe I'll bite the bullet and try this on one of my points...

**Comment by Dean2706** [comment]
I can see someone selling a single puck 2nd hand near me fairly cheap so I might give it a go this weekend

**Comment by Dean2706** [comment]
OK, so after playing around with this over the weekend and I can say that this project is defiantly a bust, but not completely without hope.

TL;DR: No.

It does work once you flash the image to a puck and you can SSH into the device. But as soon as you expose the puck to the internet an auto update kicks in and ruins the SSH access and I think even soft bricks the device. This happens when you plug an internet connection into the WAN port or if you try and adopt the device using Google Home. Luckily it only causes a soft brick and you can flash the recovery image to bring it back.
So if you are after a WIFI router without access to the internet, then technically this project works, but this is obviously not practical.

Looking into this further I can see that Galeforce essentially patches the latest OnHub recovery image that Google provides, which is 9334.41.3, this is not current and will be updated as soon as internet is given to the device. So unless someone knows where to download the most up to date version (13729.57.27 at the time of this post) of the ChromeOS image for Gale so we can patch that version and the device will not need to be updated or if anyone knows how to maintain root/SSH even after it receives an update.

**Comment by d8ahazard** [comment]
> OK, so after playing around with this over the weekend and I can say that this project is defiantly a bust, but not completely without hope.
> 
> TL;DR: No.
> 
> It does work once you flash the image to a puck and you can SSH into the device. But as soon as you expose the puck to the internet an auto update kicks in and ruins the SSH access and I think even soft bricks the device. This happens when you plug an internet connection into the WAN port or if you try and adopt the device using Google Home. Luckily it only causes a soft brick and you can flash the recovery image to bring it back. So if you are after a WIFI router without access to the internet, then technically this project works, but this is obviously not practical.
> 
> Looking into this further I can see that Galeforce essentially patches the latest OnHub recovery image that Google provides, which is 9334.41.3, this is not current and will be updated as soon as internet is given to the device. So unless someone knows where to download the most up to date version (13729.57.27 at the time of this post) of the ChromeOS image for Gale so we can patch that version and the device will not need to be updated or if anyone knows how to maintain root/SSH even after it receives an update.

Sorry, been busy as of late with another project.

BUT, I think I can help you help me help you? :P

Looks like the only way to create a recovery image for Google Wifi boxes is using the "Onhub Recovery Utility" extension for Chrome.

After some unsuccessful wiresharking, I had the bright idea to just download the extension source, which was much more productive.

Looks like, for wifi devices, the extension is downloading an "onhub_recovery.json" file from the following URL, which is likely to change in the future...but who knows?

https://dl.google.com/dl/edgedl/chromeos/recovery/onhub_recovery.json

Looking at the JSON, I see the URL for the "old" image is here:

https://dl.google.com/dl/edgedl/chromeos/recovery/chromeos_9334.41.3_gale_recovery_stable-channel_mp.bin.zip

I tried updating the URL to include the version number you specified, but I'm getting a file not found error. Gonna try playing with stuff a bit more, but I feel like this is at least a *start* into finding the latest image. ;)

**Comment by d8ahazard** [comment]
> OK, so after playing around with this over the weekend and I can say that this project is defiantly a bust, but not completely without hope.
> 
> TL;DR: No.
> 
> It does work once you flash the image to a puck and you can SSH into the device. But as soon as you expose the puck to the internet an auto update kicks in and ruins the SSH access and I think even soft bricks the device. This happens when you plug an internet connection into the WAN port or if you try and adopt the device using Google Home. Luckily it only causes a soft brick and you can flash the recovery image to bring it back. So if you are after a WIFI router without access to the internet, then technically this project works, but this is obviously not practical.
> 
> Looking into this further I can see that Galeforce essentially patches the latest OnHub recovery image that Google provides, which is 9334.41.3, this is not current and will be updated as soon as internet is given to the device. So unless someone knows where to download the most up to date version (13729.57.27 at the time of this post) of the ChromeOS image for Gale so we can patch that version and the device will not need to be updated or if anyone knows how to maintain root/SSH even after it receives an update.

I wonder if it'd be possible to SSH into the device pre-update, and then watch logs to determine the new update URL...

**Comment by Dean2706** [comment]
I went down the same track and tried to guess the download URL based on the onhub_recovery.json and on the ChromeOS recovery.conf file found here: https://dl.google.com/dl/edgedl/chromeos/recovery/recovery.conf

I think they deliberately do not advertise the latest image to stop us from messing with it, but I like your idea of watching some logs and try to capture the URL link of the update. I am not super familiar with ChromeOS so I am not sure where to begin to find these logs.

**Comment by d8ahazard** [comment]
> I went down the same track and tried to guess the download URL based on the onhub_recovery.json and on the ChromeOS recovery.conf file found here: https://dl.google.com/dl/edgedl/chromeos/recovery/recovery.conf
> 
> I think they deliberately do not advertise the latest image to stop us from messing with it, but I like your idea of watching some logs and try to capture the URL link of the update. I am not super familiar with ChromeOS so I am not sure where to begin to find these logs.

95% chance any relevant data will be dumped to *some* file in /var/log, most likely "/var/log/syslog". Looking at the scripts that galeforce runs for patching, it's just doing "echo" commands, which means that's where it should be going to.

You could probably start ssh and then tail that file to see what's happening as it updates...and if there are any errors or any other useful information in it.

Edit: Yeah, I'm almost sure of it. it looks like the patch routine is pretty simple - it detects when an update is fired, determines which partition is A and which is B and which is active, and then tries to copy itself from the old partition to the new one. All of this should be logged to syslog, as well as any issues...

Wondering if the partitions have changed...

**Comment by d8ahazard** [comment]
@Dean2706 @darthnithin @ViPeR5000, and anybody else who reads this...

I've posted over on the Google Nest Community, asking that Google release the factory image for the device. I pointed out that images used to be available, that paths seem to have changed and have been deliberately obfuscated, and that according to right-to-repair laws, we should have access to factory images for our hardware.

While I doubt this will go far, I do seem to have gotten *somebody's* attention. 

Perhaps if more people spoke up and voiced their desire for this and/or more advanced functionality, someone would listen.

https://www.googlenestcommunity.com/t5/Nest-Wifi/Several-Questions

**Comment by ViPeR5000** [comment]
> https://www.googlenestcommunity.com/t5/Nest-Wifi/Several-Questions

The page you are trying to access was not found. Please check your URL for typos and try again.

:D

**Comment by d8ahazard** [comment]
> > https://www.googlenestcommunity.com/t5/Nest-Wifi/Several-Questions
> 
> The page you are trying to access was not found. Please check your URL for typos and try again.
> 
> :D

I swear, I tried the URL before I pasted it. This should work:

https://www.googlenestcommunity.com/t5/Nest-Wifi/Several-Questions/m-p/20693?search-action-id=2922573082&search-result-uid=20693

**Comment by sqerk** [comment]
Google plans to abandon Onhubs, cut them off from cloud management access, in December.
https://support.google.com/googlenest/answer/11257354
Even if Google doesn't officially open up them up for tinkering, their active undoing of firmare mods should at least stop. And for a lot of people not being able to change a router's settings will make it garbage. There will soon be more of these routers out there for next to nothing, and more incentive for someone who wants to make their router work again to contribute to this project.

**Comment by AlexanderZhan** [comment]
China can't access google's download server,
So this will be work in china,
or iptables -t nat -I POSTROUTING 1 -d "dl.google.com" -j DROP?

**Comment by Evidlo** [comment]
OpenWRT is supported on this device now: https://openwrt.org/toh/google/wifi

**Comment by d8ahazard** [comment]
> OpenWRT is suppoted on this device now: https://openwrt.org/toh/google/wifi

LOL, I definitely should have circled back to this issue before. This is the solution I went with, and it works pretty well, all things considered. I even wrote the "SlowLED" module for it so we can get some nice light animations. :D
